bin/git-theta-filter

```diff
@@ -59,26 +68,43 @@ def clean(args):
     checkpoint_handler = checkpoints.get_checkpoint_handler()
     model_checkpoint = checkpoint_handler.from_file(sys.stdin.buffer)
 
+    update_serializer = params.get_update_serializer()
     prev_metadata = metadata.Metadata.from_commit(repo, args.file, "HEAD").flatten()
 
     async def _clean(param_keys, new_param):
         logging.debug(f"Cleaning {'/'.join(param_keys)}")
         param_metadata = prev_metadata.get(param_keys)
         new_tensor_metadata = metadata.TensorMetadata.from_tensor(new_param)
+
         # If the parameter tensor has not changed, just keep the metadata the same
-        # TODO: currently checks if rounded parameter hash has changed. Look into better parameter equality checks.
-        if param_metadata and param_metadata.tensor_metadata == new_tensor_metadata:
-            return param_keys, param_metadata
+        # TODO: Currently uses some magic numbers for LSH distance thresholds.
+        if param_metadata:
+            hasher = lsh.get_lsh()
+            hash_distance = hasher.distance(
+                param_metadata.tensor_metadata.hash, new_tensor_metadata.hash
+            )
+            # If l2 distance < 1e-7, assume this change doesn't need to be tracked
+            if hash_distance < 1e-7:
+                return param_keys, param_metadata
+            # If 1e-7 < l2 distance < 1e-5, load up old parameter and check with np.allclose
+            elif hash_distance < 1e-5:
+                update_handler = updates.get_update_handler(
+                    param_metadata.theta_metadata.update_type
+                )(update_serializer)
+                param = await update_handler.apply(
+                    param_metadata, param_keys, repo=repo, path=args.file
+                )
+                if np.allclose(param, new_param, rtol=0, atol=1e-7):
```

Comment:
I exposed both `rtol` and `atol` through `utils.EnvVarConstants` which I thought would be fine until we put some more thought into where this global repo-level configuration should go. 

Also, I kept it so that in the LSH signature distance check it still compares against `atol`. The reasoning is that the LSH signature distance approximates the L_2 distance between the two tensors and L_2 < L_inf. Thus if L_2 distance between two signatures is less than `atol`--> L_2 distance between the two tensors is likely to be less than `atol` --> L_inf distance between the tensors is less than `atol` --> `np.allclose(tensor1, tensor2, atol, rtol)` will pass.